server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailGetMethodContract.scala

```diff
@@ -5995,4 +5994,134 @@ trait EmailGetMethodContract {
            |    "header:To:asAddresses": []
            }""".stripMargin)
   }
+
+  @Test
+  def emailGetShouldReturnGroupedAddresses(server: GuiceJamesServer): Unit = {
+    val bobPath = MailboxPath.inbox(BOB)
+    server.getProbe(classOf[MailboxProbeImpl]).createMailbox(bobPath)
+    val alicePath = MailboxPath.inbox(ALICE)
+    server.getProbe(classOf[MailboxProbeImpl]).createMailbox(alicePath)
+    val message: Message = Message.Builder
+      .of
+      .setSubject("test")
+      .setSender(ANDRE.asString())
+      .setFrom(ANDRE.asString())
+      .addField(new RawField("To", "\"  user1  \" <user1@domain.tld>, \"  user2  \" <user2@domain.tld>, Friends: <user3@domain.tld>, \"user4\" <user4@domain.tld>;"))
+      .setBody("testmail", StandardCharsets.UTF_8)
+      .build
+    val messageId: MessageId = server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString, bobPath, AppendCommand.from(message))
+      .getMessageId
+
+    val response = `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(
+        s"""{
+           |  "using": [
+           |    "urn:ietf:params:jmap:core",
+           |    "urn:ietf:params:jmap:mail"],
+           |  "methodCalls": [[
+           |     "Email/get",
+           |     {
+           |       "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |       "ids": ["${messageId.serialize}"],
+           |       "properties": ["header:To:asGroupedAddresses"]
+           |     },
+           |     "c1"]]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1].list[0]")
+      .isEqualTo(
+        s"""{
+           |    "id": "${messageId.serialize}",
+           |    "header:To:asGroupedAddresses": [
+           |      {
+           |        "name": null,
+           |        "addresses": [
+           |          {
+           |            "name": "user1",
+           |            "email": "user1@domain.tld"
+           |          },
+           |          {
+           |            "name": "user2",
+           |            "email": "user2@domain.tld"
+           |          }
+           |        ]
+           |      },
+           |      {
+           |        "name": "Friends",
+           |        "addresses": [
+           |          {
+           |            "email": "user3@domain.tld"
+           |          },
+           |          {
+           |            "name": "user4",
+           |            "email": "user4@domain.tld"
+           |          }
+           |        ]
+           |      }
+           |    ]
+           |}""".stripMargin)
+  }
+
```

Comment:
What happen when I fetch a normal header with `:asGroupAddress` ?

Reply:
Same as unparsable case, an empty list?

Reply:
No I expect to still have all of the adresses (each one with a null group ;-) )

Reply:
I misunderstood your question. Shouldn't it be all addresses in one null group?